src/gh_stats/cli.py

```diff
@@ -12,29 +12,45 @@
 
 def get_repo_data(headers: dict, owner: str, repo: str) -> dict:
     """Fetches all data required for output for each repo"""
-    # TODO: Handle missing data if non-200 response
-    resp_data = requests.get(
-        f"https://api.github.com/repos/{owner}/{repo}", headers=headers
-    ).json()
 
-    views = requests.get(
+    resp_data, views, clones = None, None, None
+
+    resp = requests.get(f"https://api.github.com/repos/{owner}/{repo}", headers=headers)
+
+    if resp.status_code == 200:
+        resp_data = resp.json()
+    else:
+        print(f"Repo {repo} returned this error: ", resp.text)
+
+    views_resp = requests.get(
         f"https://api.github.com/repos/{owner}/{repo}/traffic/views", headers=headers
-    ).json()
+    )
+
+    if views_resp.status_code == 200:
+        views = views_resp.json()
+    else:
+        print(f"Repo {repo} returned this error: ", views_resp.text)
 
-    clones = requests.get(
+    clones_resp = requests.get(
         f"https://api.github.com/repos/{owner}/{repo}/traffic/clones", headers=headers
-    ).json()
+    )
 
-    return {
-        "Repo": f"{owner}/{repo}",
-        "Forks": resp_data["forks_count"],
-        "Stars": resp_data["stargazers_count"],
-        "Watchers": resp_data["watchers_count"],
-        "Clones Total": clones["count"],
-        "Clones Unique": clones["uniques"],
-        "Views Total": views["count"],
-        "Views Unique": views["uniques"],
-    }
+    if clones_resp.status_code == 200:
+        clones = clones_resp.json()
+    else:
+        print(f"Repo {repo} returned this error: ", clones_resp.text)
```

Comment:
Here the code to check the `status_code` is repeated three times. As we're likely going to be adding features, and therefore more rest calls the API, I think we should refactor this into a function whose sole purpose is to check for a valid `response` and log errors as appropriate. This will also make it far easier to implement other checks if we ever need to check for other status codes aside from 200.